src/main/java/campus/infrastructure/Ui.java

```diff
@@ -0,0 +1,132 @@
+        for (int i = 0; i < numOfTasks; i++) {
+            int listNum = i + 1;
+            Task task = taskList.getIthTaskInteger(i);
+            listOfTasks.append(String.format("%s. %s\n", listNum, task.toString()));
+        }
+
+        String message = "------------------------------\n"
+                + String.format("%s\n", listOfTasks)
+                + "------------------------------\n";
+
+        return message;
+    }
+
+    /**
+     * Prints a message when a task is marked as done
+     * @param task The Task object instance
+     */
+    public String markDone(Task task) {
+        String message = "------------------------------\n"
+                + "Nice! I've completed this task successfully:\n"
+                + String.format("%s\n", task)
+                + "------------------------------\n";
+
+        return message;
+    }
+
+    /**
+     * Prints a message when a task is marked as undone
+     * @param task The Task object instance
+     */
+    public String markUndone(Task task) {
+        String message = "------------------------------\n"
+                + "Ok, this task is still not done yet:\n"
+                + String.format("%s\n", task)
+                + "------------------------------\n";
+
+        return message;
+    }
+
+    /**
+     * Prints a message when a task is deleted
+     * @param taskList The taskList object instance
+     * @param task The task object instance
+     */
+    public String delete(TaskList taskList, Task task) {
+        int numOfTasks = taskList.getListSize();
+
+        String message = "------------------------------\n"
+                + "Noted. Task deleted successfully. I have removed the following task:\n"
+                + String.format("%s\n", task)
+                + String.format("Now you have %s task(s) in the list.\n", numOfTasks)
+                + "------------------------------\n";
+
+        return message;
+    }
+
+    /**
+     * Prints a message when a task is added to the list
+     * @param taskList The taskList object instance
+     * @param task The task object instance
+     */
+    public String add(TaskList taskList, Task task) {
+        int numOfTasks = taskList.getListSize();
+
+        String message = "------------------------------\n"
+                + "Got it. I've added this to our list of tasks:\n"
+                + String.format("added: %s\n", task.toString())
+                + String.format("Now you have %s task(s) in the list.\n", numOfTasks)
+                + "------------------------------\n";
+
+        return message;
+    }
+
+    /**
+     * Produces the error message read from the CampusException class
+     * @param e The CampusException class instance
+     */
+    public String displayErrorMessage(CampusException e) {
+        String message = "------------------------------\n"
+                + String.format("%s\n", e.getMessage())
+                + "------------------------------\n";
```

Comment:
Consider factoring out the `--------` to a constant to avoid accidental inconsistency (although looks like you switched to use JavaFX GUI already)